tests/transformers/tests/generation/test_utils.py

```diff
@@ -98,109 +107,82 @@ def _update_default_model_kwargs(self, model_kwargs):
 
     def _get_input_ids_and_config(self, batch_size=2):
         config, inputs_dict = self.model_tester.prepare_config_and_inputs_for_common()
-        input_ids = inputs_dict[self.input_name]
+        # TODO: @raushan or @gante, use `model.main_input_name` as the main input instead of relyinn on `input_ids`
+        input_ids = inputs_dict.pop(self.input_name)[:batch_size, :]
+        inputs_dict.pop("attention_mask", None)
```

Comment:
Why are you removing `attention_mask` for these tests?

Reply:
Is there a reason why we do not use `filtered_inputs_dict` similar to transformers. If the code less differentiate from upstream it is easier to maintain for other people as well

Reply:
I see, you are looking at the most recent transformer version [here](https://github.com/huggingface/transformers/blob/48831b7d1189f16a1fca9d4dad723bf4e828c041/tests/generation/test_utils.py#L104)

I was updating it based on [4.45.2](https://github.com/huggingface/transformers/blob/v4.45.2/tests/generation/test_utils.py)

I think for this function there's no harm on using the latest version. Let me try.

Reply:
There are several modifications from 4.45.2 to 4.46. We can work on that after this PR. What do you think?

Reply:
@malkomes Thank you for checking this. Let’s stick with version 4.45.2 for now. We can plan to update to 4.46 later. The next version has some other changes

Reply:
sounds good! Yeah, the 4.46 does a refactor on the tests, so it's better to work on a new PR for 4.46, and I'm happy to do it once we finish this one. (: